### PR TITLE
3840 surface sun wind exposure

### LINKED ISSUE #3840: Sun/Wind exposure defaults for walls/ceilings when OtherSideCoefficients outside boundary condition is set
Setting a surface outside boundary condition to 'OtherSideCoefficients' will reset that surface's Sun and Wind Exposure to defaults:

https://github.com/NREL/OpenStudio/blob/2c4dd5c9d790d1e1a019b328ca06957660733478/src/model/Surface.cpp#L915-L931

For surfaces other than floors, this will set them to be 'SunExposed/WindExposed':

https://github.com/NREL/OpenStudio/blob/2c4dd5c9d790d1e1a019b328ca06957660733478/src/model/Surface.cpp#L1278-L1345

EnergyPlus will ignore solar effects for surfaces with OtherSideCoefficients, but will write a warning for each surface. I'm not sure if the same is true for wind effects. However, if a user is defining OtherSideCoefficients, shouldn't their choices for sun/wind exposure be retained and not reset?

### PR DESCRIPTION
**Pull request by mpraprost**
Pull request overview
---------------------

 - Fixes #3840  (IF THIS IS A DEFECT)
 - DESCRIBE PURPOSE OF THIS PULL REQUEST

Please read [OpenStudio Pull Requests](https://github.com/NREL/OpenStudio/wiki/OpenStudio-Pull-Requests) to better understand the OpenStudio Pull Request protocol.

### Pull Request Author

Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.

 - [ ] Model API Changes / Additions
 - [ ] Any new or modified fields have been implemented in the EnergyPlus ForwardTranslator (and ReverseTranslator as appropriate)
 - [ ] Model API methods are tested (in `src/model/test`)
 - [ ] EnergyPlus ForwardTranslator Tests (in `src/energyplus/Test`)
 - [ ] If a new object or method, added a test in NREL/OpenStudio-resources: Add Link
 - [ ] If needed, added VersionTranslation rules for the objects (`src/osversion/VersionTranslator.cpp`)
 - [ ] Checked behavior in OpenStudioApplication, adjusted policies as needed (`src/openstudio_lib/library/OpenStudioPolicy.xml`)
 - [ ] Verified that C# bindings built fine on Windows, partial classes used as needed, etc.
 - [x] All new and existing tests passes
 - [ ] If methods have been deprecated, update rest of code to use the new methods

**Labels:**

 - [ ] If change to an IDD file, add the label `IDDChange`
 - [ ] If breaking existing API, add the label `APIChange`
 - [x] If deemed ready, add label `Pull Request - Ready for CI` so that CI builds your PR

### Review Checklist

This will not be exhaustively relevant to every PR.
 - [ ] Perform a Code Review on GitHub
 - [ ] Code Style, strip trailing whitespace, etc.
 - [ ] All related changes have been implemented: model changes, model tests, FT changes, FT tests, VersionTranslation, OS App
 - [ ] Labeling is ok
 - [ ] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] If feature, test running new feature, try creative ways to break it
 - [ ] CI status: all green or justified

### CODE REVIEW

File: src/model/Surface.cpp [hunk 1]
```diff
@@ -924,8 +924,6 @@ namespace detail {
     if (test){
       test = this->setString(OS_SurfaceFields::OutsideBoundaryCondition, "OtherSideCoefficients");
       OS_ASSERT(test);
-      this->assignDefaultSunExposure();
```
Comment (macumber) [inline src/model/Surface.cpp]: @mpraprost @kbenne  I think this one might be better fixed by adding "OtherSideCoefficients" to the list of cases checked in `assignDefaultSunExposure` and `assignDefaultWindExposure`.  This should be in the same case as if it is `Surface`.

https://github.com/NREL/OpenStudio/blob/e89a74a0178ee2e42c6c74d8f44af8f9d5510deb/src/model/Surface.cpp#L1276

Comment (macumber) [reply src/model/Surface.cpp]: Also glad to see you getting in the C++ Marley :-)

Comment (mpraprost) [reply src/model/Surface.cpp]: I think we have two potential solutions to this issue. When OtherSideCoefficients is specified, we can either:
1. not reset the sun/wind exposure settings at all (my original commit)
2. reset the sun/wind exposure to 'NoWind' and 'NoSun' (result of making the changes Dan suggested)

Do you have any input on this? Based on the original issue description, it sounded like @eringold would like to retain the original sin/wind exposure settings. @macumber do you have other insight into why solution 2 might be better?

Comment (eringold) [reply src/model/Surface.cpp]: I actually think resetting to 'NoSun' at least would be fine. 

This came about due to perhaps another issue, which is that the Sketchup plugin allows you to set a surface outside boundary condition to 'OtherSideCoefficient' *without assigning a SurfaceProperty:OtherSideCoefficients object to that surface*, but you can't through the App or the sdk: `setOutsideBoundaryCondition` will first look for a surface property object assigned to the surface: https://github.com/NREL/OpenStudio/blob/2c4dd5c9d790d1e1a019b328ca06957660733478/src/model/Surface.cpp#L457-L459.

So I would go through my model in the plugin, set the surface boundaries to OtherSideCoefficients and make sure the sun/wind exposures are set, and then create and apply a surface property object to those surfaces with a measure, only to find that my exposures were reset!

Anyway, the EnergyPlus documentation is clear that solar effects aren't considered for OtherSideCoefficients, and like I said it will throw a warning, so at least set that to 'NoSun'. The SurfaceProperty:OtherSideCoefficients has an input for Wind Speed Coefficient, so presumably the wind effects are retained, so maybe don't reset the wind exposure, and leave that assignment up to the user.

Comment (macumber) [reply src/model/Surface.cpp]: My suggestion was mostly just to be consistent with the pattern of how other surface types work.  It seems pretty clear that the OtherSideCoefficients option was added and we just forgot to update the cases for it.

Comment (jmarrec) [reply src/model/Surface.cpp]: I agree with @macumber on this one. Can you adjust @mpraprost or let me know if you don't have the bandwidth please? 

If you do have time to adjust, please re-title the PR accordingly so it's understanble in a standalone changelog context, something like 

> Reset Surface Wind/SunExposure to NoSun/NoWind when setting Outside Boundary Condition to `OtherSideCoefficients`

**Comment by jmarrec** [comment]
Additional changes are needed on this test:

```
2063: [ RUN      ] ModelFixture.SurfacePropertyOtherSideCoefficients
2063: C:\src\OpenStudio\src\model\test\Surface_GTest.cpp(865): error: Expected equality of these values:
2063:   "SunExposed"
2063:   wall1.sunExposure()
2063:     Which is: "NoSun"
2063: C:\src\OpenStudio\src\model\test\Surface_GTest.cpp(866): error: Expected equality of these values:
2063:   "WindExposed"
2063:   wall1.windExposure()
2063:     Which is: "NoWind"
2063: C:\src\OpenStudio\src\model\test\Surface_GTest.cpp(870): error: Expected equality of these values:
2063:   "SunExposed"
2063:   wall1.sunExposure()
2063:     Which is: "NoSun"
2063: C:\src\OpenStudio\src\model\test\Surface_GTest.cpp(871): error: Expected equality of these values:
2063:   "WindExposed"
2063:   wall1.windExposure()
2063:     Which is: "NoWind"
```

**Comment by jmarrec** [comment]
Fixed in #3916
